File: mobile/lib/domain/repository/debt_repository.dart
import '../model/debt.dart';

abstract class DebtRepository {
  Future<void> create(Debt debt);
  Future<Debt?> findById(String id);
  Future<List<Debt>> listByFamily(String familyId, {bool openOnly = false});
  Future<void> update(Debt debt);
  Future<void> softDelete(String id);
  Stream<List<Debt>> watchByFamily(String familyId);
}

File: backend/src/main/kotlin/mymoney/domain/repository/DebtRepository.kt
package mymoney.domain.repository

import mymoney.domain.model.Debt
import java.util.UUID

interface DebtRepository {
    suspend fun create(debt: Debt): Debt
    suspend fun findById(id: UUID): Debt?
    suspend fun list(familyId: UUID, openOnly: Boolean = false): List<Debt>
    suspend fun update(debt: Debt): Debt
    suspend fun softDelete(id: UUID)
}
